torch/utils/data/_utils/collate.py

```diff
@@ -158,6 +158,11 @@ def collate_tensor_fn(batch, *, collate_fn_map: Optional[Dict[Union[Type, Tuple[
             "Batches of nested tensors are not currently supported by the default collate_fn; "
             "please provide a custom collate_fn to handle them appropriately."
         )
+    if elem.layout in {torch.sparse_coo, torch.sparse_csr, torch.sparse_bsr, torch.sparse_csc, torch.sparse_bsc}:
```

Comment:
Is this set of sparse types available elsewhere - curious to know if this can expand in the future and would require updating all instances of such sets?

Reply:
No, we have not created a public API that would provide all sparse layouts as a set.

While various tools have implemented such a set (e.g., https://github.com/pytorch/pytorch/blob/c6f435befd633ae3fecf52c597c78f0934c7537c/torch/sparse/__init__.py#L522), they are typically private to the corresponding tools.

On the other hand, if there will be a new sparse layout added (which btw, is a very rare event considering the high threshold of introducing new sparse layouts), relying on such a set of sparse layouts here would not avoid the need to update dataloader support for new sparse layouts because the support relies on the details of sparse tensor implementations.